Gravity-Notes/src/Components/Notes-card/Notecard.js

```diff
@@ -9,40 +9,68 @@ import { useAuth } from "../../Contexts/Auth-context"
             <small>{note.data}</small>
             </div>
             <p>{note.content}</p>
-            <div className="cardicon">
-                <button><MdMoreVert size={25} /></button>
-                <button onClick={() => deleteNote(note,authState,noteDispatch) }><MdDelete size={25} /></button>
-                <button onClick={() => editHandler(note)}><BiEditAlt size={25} /></button>
-                <button onClick={() => archiveNote(note,authState,noteDispatch) }><BiArchiveIn size={25} /></button>
-                <button><HiOutlineUserAdd size={24} /></button>
-                <button><BiBellPlus size={25} /></button>
+            <div className={`cardicon ${note.color}`}>
+                <button className="card-action-btn"><MdMoreVert size={25} /></button>
+                <button className="card-action-btn" onClick={() => deleteNote(note,authState,noteDispatch) }><MdDelete size={25} /></button>
+                <button className="card-action-btn" onClick={() => editHandler(note)}><BiEditAlt size={25} /></button>
+                <button className="card-action-btn" onClick={() => archiveNote(note,authState,noteDispatch) }><BiArchiveIn size={25} /></button>
+                <button className="card-action-btn" onClick={ ()=> showColorSelector(note)} ><IoColorPaletteSharp size={24} /></button>
+                <button className="card-action-btn"><BiBellPlus size={25} /></button>
             </div>
-        </div>
+        
+
+         {/* COLOR SELECTOR BOX */}
+      {colorSelector
+        && 
+       <div className="card-color-box">
+         <span className="card-pointer">^</span>
+       <button onClick={()=>applyColorOnCard("",note)} className="card-color-btn"> <MdOutlineFormatColorReset size={5}/>  </button>
+       <button onClick={()=>applyColorOnCard("red",note)} className="card-color-btn red"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("orange",note)} className="card-color-btn orange"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("yellow",note)} className="card-color-btn yellow"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("green",note)} className="card-color-btn green"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("purple",note)} className="card-color-btn purple"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("pink",note)} className="card-color-btn pink"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+       <button onClick={()=>applyColorOnCard("brown",note)} className="card-color-btn brown"> <MdOutlineFormatColorReset size={5}  visibility={"hidden"} />  </button>
+     </div> }
+     </div>
```

Comment:
you can make array of color name and return buttons.

Reply:
thanks, @Jay-Gadhiya j and i will refactor my CSS Soon first going with all  adding all functionality

Reply:
You can create a separate component for color pallete. Just a suggestion.